PWG/EMCAL/EMCALtasks/AliEmcalAodTrackFilterTask.cxx

```diff
@@ -247,21 +235,32 @@ void AliEmcalAodTrackFilterTask::UserExec(Option_t *)
       return;
     }
   }
-
+*/
+AliParticleContainer* OutputCont= static_cast<AliParticleContainer*>(fParticleCollArray.Last());
```

Comment:
This is a strong assumption on which particle container is the embedded container. If the embedded is added first (which would be a valid choice), this won't work unexpectedly. You could loop over `fParticleCollArray` and check all containers if any of them are embedded. Or alternatively, does it really needed to be attached to the embedded event? Could it also be fine in the main event? (I didn't review closely enough to answer - just something to consider)

Reply:
Indeed, there is certainly a better way to pick the correct event from the fParticleCollArray. However, isn't fParticleCollArray a member of only the filter task? Doesn't each analysis task have its own  fParticleCollArray? Because then the order of the containers for this task isn't very user-dependent since first container corresponds to the input to the filter task and second or last to the output of the filter task. This should work for both "normal" and embedded analysis. Did I misunderstand the nature of the object?

I try to avoid adding an extra case for the embedded even though that case is the main motivation behind the changes. The filtered output is assigned to a container which is written either to the main or the embedded event. Given the dependence of the jet finders to the supplied track container name, if one wants to cluster jets at the hybrid level for an embedded event then the filter task will have to be run twice ,once for the embedded and once for the main event. By having the output written in separate event records you exclude cases of overwriting/replacing tracks. Is my reasoning clear or am I missing something?